include/SFML/Graphics/Shader.hpp

```diff
@@ -739,8 +739,8 @@ class SFML_GRAPHICS_API Shader : GlResource, NonCopyable
     ////////////////////////////////////////////////////////////
     // Types
     ////////////////////////////////////////////////////////////
-    typedef std::map<int, const Texture*> TextureTable;
-    typedef std::map<std::string, int> UniformTable;
+    typedef std::unordered_map<int, const Texture*> TextureTable;
+    typedef std::unordered_map<std::string, int> UniformTable;
```

Comment:
I had another look at this. One thing that looks a bit suspicious is the following code:
https://github.com/SFML/SFML/blob/eef309e870f523363339e02b95d4f6eb2a18b429/src/SFML/Graphics/Shader.cpp#L971-L985

We iterate over the textures with both index and iterator, and assign a uniform based on the index.
I'm not sure if the actual index value matters, but this is an example where `unordered_map` might change semantics.

Reply:
I don't think this would actually change anything in practice, would it? As far as I can tell, the indexes are not really used anywhere else, they're just being used as unique identifiers for the texture units. Am I missing something?

Reply:
You're probably right, but I haven't touched this code for a while, so I thought I'd highlight it for review. Maybe @binary1248 knows this?

Reply:
@vittorioromeo is right. We just have to assign each texture to the next free texture unit. The ordering shouldn't make a difference. Unless the user changes `m_textures` often there should be no performance penalties either.

Reply:
OK, I am going to keep this change. Thank you for confirming.